worker/uniter/runner/context/context.go

```diff
@@ -1061,13 +1075,42 @@ func (ctx *HookContext) GrantSecret(uri *coresecrets.URI, args *jujuc.SecretGran
 			return ErrIsNotLeader
 		}
 	}
-	ctx.secretChanges.grant(uniter.SecretGrantRevokeArgs{
+	uniterArg := uniter.SecretGrantRevokeArgs{
 		URI:             uri,
-		ApplicationName: args.ApplicationName,
-		UnitName:        args.UnitName,
-		RelationKey:     args.RelationKey,
+		ApplicationName: arg.ApplicationName,
+		UnitName:        arg.UnitName,
+		RelationKey:     arg.RelationKey,
 		Role:            coresecrets.RoleView,
-	})
+	}
+	params := uniterArg.ToParams()
+	if len(params.SubjectTags) != 1 {
+		return errors.NotValidf("subject tags %v", params.SubjectTags)
```

Comment:
Sorry, I should have suggested this the firs time. We are checking how many - we want 1 - so the error message should reflect that.

NewNotValidf(nil, "expected only 1 subject, got %d", len(params.SubjectTags))